### ISSUE TITLE
Interleaving XML elements support

### ISSUE THREAD

**Issue by niyue** [drmohundro/SWXMLHash#19]
Hi David, 

Thanks for the library, I like the API of this library but have one feature request (or bug report?) on it.

In some XML documents, elements with the same name can appear in any order instead of appear continuously. For example, an XHTML document:
```
<html>
  <body>
    <p>one</p>
    <div>two</div>
    <p>three</p>
    <div>four</div>
  </body>
</html>
```

As you can see, tag p and tag div can interleave. And to parse and render this XHTML correctly, all the elements order in this document should be kept. So one-two-three-four should be shown in the browser instead of one-three-two-four. However, this seems not possible in current implementation. 

Right now, in current implementation, all child elements are stored in an ordered dictionary, whose key is the element name and value is an array of elements with the same element name. When adding an element, the new element will be appended to the dictionary entry with the element name, and when retrieving all children, the entire dictionary will be iterated entry by entry.

Is there any plan to support the XML document with interleaving pattern like above? The current implementation seems to me incorrect when handling such interleaving pattern by giving me incorrect elements order because it will groups elements by their names first. 

To support this interleaving pattern, we could simply use an array to replace the ordered dictionary, and this will not only make the implementation correct (XML document is essentially a linear document I think) but also much simpler. The only down side is the subscript access by element name will be slower but that may be acceptable. What do you think?

**Comment by drmohundro** [comment]
Great thought - the dictionary approach did make a few assumptions that were incorrect. Also, regarding the speed, I'm working on an optional lazy loading approach, which should mean that the speed shouldn't be a big deal, because it will only have the elements indexed that were requested.

**Comment by drmohundro** [comment]
Thanks for the suggestion regarding just using an array @niyue - that's simpler and handles both scenarios. I published 0.6.4 with this fix.

**Comment by niyue** [comment]
Wow, you are fast. Thanks so much, I checked it out using the xcode-6.3 branch, and it is working very well in my project.
